doc/conf.py

```diff
@@ -1,161 +1,12 @@
-import os
+# Obtain shared config values
 import sys
-import time
+from pathlib import Path
 
-# Configuration file for the Sphinx documentation builder.
+DOC_ROOT = Path(__file__).parent.absolute()
+sys.path.extend([DOC_ROOT.as_posix(), DOC_ROOT.parent.as_posix()])
 
-# -- Project information
-project = "Nuitka the Python Compiler"
-copyright = "%s, Kay Hayen and Nuitka Contributors" % time.gmtime().tm_year
-author = "Kay Hayen"
-release = version = ""
-
-# For autodoc to work
-sys.path.append(os.path.join(os.path.dirname(__file__), ".."))
-from update import importNuitka
-importNuitka()
-del sys.path[-1]
-
-# -- General configuration
-
-extensions = [
-    # Sphinx's own extensions
-    "sphinx.ext.autodoc",
-    "sphinx.ext.extlinks",
-    "sphinx.ext.intersphinx",
-    "sphinx.ext.mathjax",
-    "sphinx.ext.todo",
-#    "sphinx.ext.viewcode",
-    # External extensions
-    "sphinx_copybutton",
-    "sphinx_design",
-#    "sphinx_inline_tabs",
-    "sphinxcontrib.youtube",
-    "sphinx-favicon",
-    "sphinx_sitemap",
-    # Disabled, loads to slow and cannot be limited to needing pages.
-    # "sphinxcontrib.asciinema",
-    # Blog extension
-    "ablog",
-]
-
-intersphinx_mapping = {
-    "python": ("https://docs.python.org/3/", None),
-    "sphinx": ("https://www.sphinx-doc.org/en/master/", None),
-}
-intersphinx_disabled_domains = ["std"]
-
-templates_path = ["_templates"]
-
-# Options for ABlog
-
-# The "title" for the blog, used in active pages.  Default is ``'Blog'``.
-blog_title = "Nuitka Blog"
-
-# Base URL for the website, required for generating feeds.
-# e.g. blog_baseurl = "http://example.com/"
-blog_baseurl = "https://nuitka.net"
-blog_path = "blog"
-blog_feed_fulltext = True
-blog_feed_archives = True
-blog_feed_length = None
-
-# RTD theme has fontawesome used, allow ablog to use it too.
-fontawesome_included = True
-
-
-# TODO: Optimized feeds one day?
-x_blog_feed_templates = {
-    # Use defaults, no templates
-    "atom": {},
-    # Create content text suitable posting to social media
-    "social": {
-        # Format tags as hashtags and append to the content
-        "content": "{{ title }}{% for tag in post.tags %}"
-        " #{{ tag.name|trim()|replace(' ', '') }}"
-        "{% endfor %}",
-    },
-}
+from shared_conf import *
```

Comment:
I actually thought this was part of the last merge, the shared conf, so this had me confused initially. I am not an expert on `pathlib`, because most code I write is now allowed to use it, not here, this can and should be all modern code. But using `as_posix()` feels wrong, since it sounds platform specific.

Reply:
`pathlib` 是 is a Python built-in library that provides a more Pythonic interface. `as_posix()` simply changes the path to a Unix-style string.

Reply:
I am saying you ought to not have to do that, but ask for something that uses a Windows path on that platform, and something else on non-POSIX. I believe that ought to be a wrong use of it. Checking the docs of `pathlib`, it is not obvious, what to pass on, but in the code there is this:

```
    def __str__(self):
        """Return the string representation of the path, suitable for
        passing to system calls."""
```

So I would assume that rather than `path.as_posix()` it should be `str(path)` or, I guess, usage in a format string normally just works as well. The `str` will be the portable code for sure, and `as_posix` might work, but has no promise to do so.